nix/README.md

```diff
@@ -0,0 +1,102 @@
+
+> There are, of course, exceptions to this general principle. macOS system settings for example are written via `defaults write` which updates a SQLite (TODO: double check this) database.
+
+The complexity of a dotfiles configuration comes primarily from the sheer variety of files/expected locations and the occasional need to interface with a third party tool to move a given file to its corresponding location. For example, to install a given package one needs to interface with a package manager (`apt`, `yum`, `brew`, etc.).
+
+Nix tooling abstracts this complexity away by providing a consistent interface to declare a wider range of config than any other tool I'm aware of. This varies from the least abstract symlinking of arbitrary files to setting high-level keys in a corresponding module which Nix tooling realizes by writing the corresponding files.
+
+In the high-level key case the Nix ecosystem has created a wrapper around a bit of config that makes declaring it as simple as setting the value of a specific key. For example, this [`system.defaults.NSGlobalDomain._HIHideMenuBar`](TODO: fill this) key controls whether or not the macOS menu bar is hidden. This [`users.users.nathan.packages`](TODO: fill this) key specifies the set of packages available to the `nathan` user. Nix tooling performs the actions required to apply this config: writing files, installing a package, flipping a macOS setting, etc.
+
+In other situations it may make more sense to declare a bit of config in its own file. To apply this config the file needs to be moved to the expected location for it. For example, [this key](TODO: fill this) causes Nix to apply the [`tmux.conf`](../tmux/tmux.conf) config in these dotfiles by moving it to `~/.tmux.conf`.
+
+> Arbitrary config code can also be inlined in `.nix` files via the """...""" syntax. See [here](TODO: fill this) for example. Nix tooling will then write this config into a file in the expected location for the corresponding tool.
+
+## Nix ecosystem
+
+The Nix umbrella includes multiple projects that when combined approach this goal better than any other system I am currently aware of.
+
+These projects are:
+
+- the Nix language tuned specifically for declarative, replicable configuration
+- the Nixpkgs package manager
+- Home Manager - a tool for managing the user environment (shell, editor, CLI tools, etc.)
+- OS-specific tooling
+  - NixOS - an operating system built completely atop Nix thus supporting perfectly replicable, declarative configuration
```

Comment:
are you going to use NixOS later? only mention if it'll be relevant i think

Reply:
also "perfectly" is flowery - recommend against too flowery language (this is a readme vs. a sales pitch lol)